### PR TITLE
Jupyter Plugin: Add resource limit spec to pipelines

### PR DESCRIPTION
**Pull request by keramosg**
Replacement for #9107 and #9110 with internal branch.

---

Adds method for adding GPUs (resource_limits spec) to pipelines spawned by the jupyter extension.

[INT-1048]

[INT-1048]: https://pachyderm.atlassian.net/browse/INT-1048?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: jupyter-extension/jupyterlab_pachyderm/tests/test_integrations.py [hunk 1]
```diff
@@ -424,6 +424,7 @@ def _update_metadata(notebook: Path, project_name: str, repo_name: str, pipeline
     config.pipeline = dict(name=pipeline_name, project=dict(name=project_name))
     # sub in repo_name
     config.input_spec = f"pfs:\n  repo: {repo_name}\n  glob: \"/*\""
+    config.resource_spec = "" # this is currently not being tested so it is set to the empty string
```
Comment (smalyala) [inline jupyter-extension/jupyterlab_pachyderm/tests/test_integrations.py:427]: If we're not testing now, let's create a ticket to add tests later?

**Review by smalyala — APPROVED** [review]
LGTM

**Review by chainlink — APPROVED** [review]
Tested this out and it works as expected for the simple and advanced flows

File: jupyter-extension/jupyterlab_pachyderm/tests/test_integrations.py (hunk 1 above)
Comment (chainlink) [reply jupyter-extension/jupyterlab_pachyderm/tests/test_integrations.py:427]: Done! https://pachyderm.atlassian.net/browse/INT-1057

**Pull request merged**